lib/truffle/thread.rb

```diff
@@ -29,4 +29,25 @@ class ThreadError < StandardError
 class Thread
   Queue = ::Queue
   SizedQueue = ::SizedQueue
+
+  class Queue
+    def pop(non_block = false, timeout: nil)
+      Primitive.queue_pop(self, non_block, validate_and_prepare_timeout(non_block, timeout))
+    end
+    alias_method :shift, :pop
+    alias_method :deq, :pop
+
+    private def validate_and_prepare_timeout(non_block, timeout)
+      if non_block
+        raise ArgumentError, "can't set a timeout if non_block is enabled" unless Primitive.nil?(timeout)
+      else
+        unless Primitive.nil?(timeout)
+          # Convert seconds to milliseconds.
+          return (Truffle::Type.rb_num2dbl(timeout) * 1000).to_i
+        end
+      end
+
+      timeout
+    end
+  end
```

Comment:
**subjective:** I would consider moving such private helpers to a proper `...Operations` helper module (there is no such one for Queue class yet).

They are located in https://github.com/oracle/truffleruby/tree/master/src/main/ruby/truffleruby/core/truffle, e.g. `enumerable_operations.rb`

Reply:
Ah yes, that makes more sense, I'll do the same as in https://github.com/oracle/truffleruby/pull/3047/files#diff-25c4fa0641c1eb5f7d0c9cce861bb5ae48f7c7ad45a1aae82c7576c4dd0eb9ed